# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1499523272 14400
#      Sat Jul 08 10:14:32 2017 -0400
# Node ID 6be1bf9455e3431f1ea3ef1bdaa5ea33bf4446be
# Parent  0ebc4f1a57145d971b47693171394a3503cfe76e
fix argument checks for internal profiler functions

* profiler.cc (F__profiler_enable__, F__profiler_reset__,
F__profiler_data__): Error with usage message for incorrect number
of arguments.

diff --git a/libinterp/parse-tree/profiler.cc b/libinterp/parse-tree/profiler.cc
--- a/libinterp/parse-tree/profiler.cc
+++ b/libinterp/parse-tree/profiler.cc
@@ -387,44 +387,44 @@ DEFUN (__profiler_enable__, args, ,
 Undocumented internal function.
 @end deftypefn */)
 {
   int nargin = args.length ();
 
   if (nargin > 1)
     print_usage ();
 
-  if (nargin > 0)
+  if (nargin == 1)
     profiler.set_active (args(0).bool_value ());
 
   return ovl (profiler.is_active ());
 }
 
 // Clear all collected profiling data.
 DEFUN (__profiler_reset__, args, ,
        doc: /* -*- texinfo -*-
 @deftypefn {} {} __profiler_reset__ ()
 Undocumented internal function.
 @end deftypefn */)
 {
-  if (args.length () > 0)
-    warning ("profiler_reset: ignoring extra arguments");
+  if (args.length () != 0)
+    print_usage ();
 
   profiler.reset ();
 
   return ovl ();
 }
 
 // Query the timings collected by the profiler.
 DEFUN (__profiler_data__, args, nargout,
        doc: /* -*- texinfo -*-
 @deftypefn {} {} __profiler_data__ ()
 Undocumented internal function.
 @end deftypefn */)
 {
-  if (args.length () > 0)
-    warning ("profiler_data: ignoring extra arguments");
+  if (args.length () != 0)
+    print_usage ();
 
   if (nargout > 1)
     return ovl (profiler.get_flat (), profiler.get_hierarchical ());
   else
     return ovl (profiler.get_flat ());
 }
